### PR TITLE
Restructure and codegen

### PR DESCRIPTION
**Pull request by colinhacks**
- [x] Add `--file` mode
- [x] Finalize command for QB generation (currently `npx @edgedb/generate edgeql-js`)

For the moment, I've disabled CI based releasing to avoid any accidental publishing in advanced of the `edgedb-js` 1.0 release. We can release 1.0 manually and re-introduce down the line.

I also removed `--watch` mode for the moment. It could be implemented later, but third-party tools like `nodemon` are good enough for the moment. There's no decent file watcher package that exists on both NPM and deno.land/x which complicated the implementation.

### CODE REVIEW

File: packages/generate/src/queries.ts [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+  // }
+
+  // find all *.edgeql files
+  // for query in queries:
+  //   generate output file
+}
+
+async function getMatches(root: string) {
+  return adapter.walk(root, {
+    match: [/\.edgeql$/],
```
Comment (Sikarii) [inline packages/generate/src/queries.ts:98]: Could this match a file without a name (`.edgeql`) and cause issues later down the line?

File: packages/generate/test/queries/tsconfig.build.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
+  "extends": "./tsconfig.json",
+  "compilerOptions": {"allowJs": false},
+  "include": [],
+  "files": [
+    "test.ts", "getMoviesStarring.edgeql.ts"
```
Comment (Sikarii) [inline packages/generate/test/queries/tsconfig.build.json:6]: Is this missing freeShape.edgeql.ts?

**Review by Sikarii — COMMENTED** [review]
Was slightly painful to review with all the dangling comma changes, but got there regardless.
I spotted a few things that seemed out of place to me, hope this helps.

Additionally, is it worth making the JS/TS client docs to be consistent with the dangling comma changes?

**Comment by Sikarii** [comment]
Should the generators have their own directory in the default dir (`dbschema`), say `dbschema/generated/<generator>`?

This would prevent bloating the dbschema directory if we're gonna have numerous generators.
And in addition, this would make ignoring (for VCS and such) all the generated code very easy.

Lastly, what's the stance on supporting pluggable generators?
From what I understood, all the generators live in the EdgeDB-owned package, is there any room to have a plugin system of sorts?

File: docs/generation.rst [hunk 3]
```diff
@@ -1,266 +1,139 @@
...
-  Checking the generated query builder into version control
-  is NOT RECOMMENDED. Would you like to update .gitignore to ignore
-  the query builder directory? The following line will be added:
+**Node.js**
 
-  dbschema/edgeql-js
+.. code-block:: bash
 
-  [y/n] (leave blank for "y")
+  $ npx edgedb generate <generator> [options]
```
Comment (Sikarii) [inline docs/generation.rst:76]: Shouldn't this command be `npx @edgedb/generate <generator> [options]`?

File: docs/queries.rst [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+      └── getUser.edgeql
+
+
+The following command will run the ``queries`` generator.
+
+.. code-block:: bash
+
+  $ npx @edgedb/generate queries
+
+The generator will detect the project root by looking for an ``edgedb.toml``, then scan the directory for ``*.edgeql`` files. In this case, there's onlny one: ``queries/getUser.edgeql``.
```
Comment (Sikarii) [inline docs/queries.rst:55]:
```diff
- onlny
+ only
```

File: docs/queries.rst [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+  ├── edgedb.toml
+  ├── index.ts
+  ├── dbschema
+  │   └── queries.ts  <-- generated file
+  └── queries
+      └── getUser.edgeql
+      └── getMovies.edgeql
+
+
+We can now use this function in our code.
```
Comment (Sikarii) [inline docs/queries.rst:161]:
```diff
- use this function
+ use these functions
```

File: docs/querybuilder.rst [hunk 6]
```diff
@@ -145,75 +106,82 @@ Then create ``import_map.json`` with the following contents. Both lines must be
     }
   }
 
-Optionally, you can specify a version:
+Version control
+^^^^^^^^^^^^^^^
 
-.. code-block:: json
+The first time you run generator, you'll be prompted to add the generated
```
Comment (Sikarii) [inline docs/querybuilder.rst:112]:
```diff
- The first time you run generator
+ The first time you run the generator
```

File: docs/querybuilder.rst [hunk 7]
```diff
@@ -306,6 +295,60 @@ extracted with the ``$infer`` helper.
...
+  }
+
+  interface Movie {
+    id: string;
+    title: string;
+    genre?: Genre | null;
+    actors: Person[];
+  }
+
+Any types declared in a non-``default`` module  will be generated into an
```
Comment (Sikarii) [inline docs/querybuilder.rst:349]:
```diff
- module  will
+ module will

**Review by Sikarii — COMMENTED** [review]
Some nitpicks about the docs.
There were also some places where `.` was used in place of `:` (when listing things).

File: packages/generate/src/codecToType.ts [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+function generateSetType(type: string, cardinality: Cardinality): string {
+  switch (cardinality) {
+    case Cardinality.MANY:
+      return `${type}[]`;
+    case Cardinality.ONE:
+      return type;
+    case Cardinality.AT_MOST_ONE:
+      return `${type} | null`;
+    case Cardinality.AT_LEAST_ONE:
+      return `AtLeastOne<${type}>`;
```
Comment (jaclarke) [inline packages/generate/src/codecToType.ts:85]: We need to import this `AtLeastOne` type when it's used.

Comment (colinhacks) [reply packages/generate/src/codecToType.ts]: I inlined this instead

**Comment by soilSpoon** [comment]
May I know when those changes will be released as npm packages?

**Pull request merged**